### PR TITLE
Refactored code by formatting code removing unused imports and debug …

### PR DESCRIPTION
**Pull request by piuswalter**
…statements

### CODE REVIEW

File: src/Exercise03AI.ipynb [hunk 1]
```diff
@@ -58,27 +59,29 @@
...
     "\n",
+    "\n",
     "class Exercise03AI(Exercise02AI):\n",
     "    \"\"\"Chooses middle game moves using minimax algorithm and piece square tables\"\"\"\n",
+    "\n",
     "    def __init__(self, **kwargs) -> None:\n",
     "        super().__init__(**kwargs)\n",
     "        self.is_king_endgame: bool = False\n",
-    "        # Create pice square table mapping (needs to revert tables) \n",
+    "        # Create pice square table mapping (needs to revert tables)\n",
```
Comment (p-fruck) [inline src/Exercise03AI.ipynb:69]:
```suggestion
    "        # Create piece square table mapping (needs to revert tables)\n",
```

File: src/Exercise03AI.ipynb [hunk 2]
```diff
@@ -190,17 +195,21 @@
...
     "            (\n",
-    "                not board.pieces(chess.QUEEN, color)\n",
-    "                or len(board.pieces(chess.BISHOP, color)) + len(board.pieces(chess.KNIGHT, color)) < 2 \n",
-    "                and not board.pieces(chess.ROOK, color)\n",
+    "                (\n",
+    "                    not board.pieces(chess.QUEEN, color)\n",
+    "                    or len(board.pieces(chess.BISHOP, color))\n",
+    "                    + len(board.pieces(chess.KNIGHT, color))\n",
+    "                    < 2\n",
+    "                    and not board.pieces(chess.ROOK, color)\n",
```
Comment (p-fruck) [inline src/Exercise03AI.ipynb:208]: I don't like

File: src/Exercise03AI.ipynb [hunk 3]
```diff
@@ -518,9 +568,15 @@
...
-    "assert unit_test_player.tables == unit_test_player.PIECE_TO_PST, \"Reset was not successfull! (tables)\""
+    "assert (\n",
+    "    unit_test_player.last_evaluation is None\n",
+    "), \"Reset was not successfull! (last_evaluation)\"\n",
+    "assert (\n",
+    "    unit_test_player.is_king_endgame is False\n",
+    "), \"Reset was not successfull! (is_king_endgame)\"\n",
+    "assert (\n",
+    "    unit_test_player.tables == unit_test_player.PIECE_TO_PST\n",
+    "), \"Reset was not successfull! (tables)\""
```
Comment (p-fruck) [inline src/Exercise03AI.ipynb:579]:
```suggestion
    "    unit_test_player.last_evaluation is None\n",
    "), \"Reset was not successful! (last_evaluation)\"\n",
    "assert (\n",
    "    unit_test_player.is_king_endgame is False\n",
    "), \"Reset was not successful! (is_king_endgame)\"\n",
    "assert (\n",
    "    unit_test_player.tables == unit_test_player.PIECE_TO_PST\n",
    "), \"Reset was not successful! (tables)\""
```

**Review by antonplagemann — APPROVED** [review]

**Review by p-fruck — CHANGES_REQUESTED** [review]

File: src/Exercise03AI.ipynb (hunk 2 above)
Comment (piuswalter) [reply src/Exercise03AI.ipynb]: Fixed in cee25b1.

**Review by p-fruck — APPROVED** [review]

**Pull request merged**